### PR TITLE
migrate some `rustc_borrowck` diagnostic

### PR DESCRIPTION
**Pull request by AndyJado**
@bors r+ rollup

it's my first pull request ever, I got all the time and energy, but I wonder if I'm heading the right place.🧎

### CODE REVIEW

**Comment by davidtwco** [comment]
You don't need to invoke `@bors` in a PR description, and only approved reviewers can do a `@bors r+` to approve something for merge. :) You might have meant `r? @username` or `r? rust-lang/<group>` to assign a reviewer.

File: compiler/rustc_borrowck/src/session_diagnostics.rs [hunk 1]
```diff
@@ -1,7 +1,15 @@
-use rustc_macros::{SessionDiagnostic, SessionSubdiagnostic};
+use rustc_macros::{LintDiagnostic, SessionDiagnostic, SessionSubdiagnostic};
 use rustc_middle::ty::Ty;
 use rustc_span::Span;
 
+#[derive(SessionDiagnostic)]
+#[error(borrowck::move_borrowed, code = "E0505")]
```
Comment (davidtwco) [inline compiler/rustc_borrowck/src/session_diagnostics.rs:6]: This isn't used anywhere, right?

File: compiler/rustc_borrowck/src/region_infer/opaque_types.rs [hunk 2]
```diff
@@ -639,17 +641,11 @@ impl<'tcx> TypeFolder<'tcx> for ReverseMapper<'tcx> {
...
-                            .struct_span_err(
-                                self.span,
-                                &format!(
-                                    "const parameter `{}` is part of concrete type but not \
-                                          used in parameter list for the `impl Trait` type alias",
-                                    ct
-                                ),
-                            )
-                            .emit();
+                        //FIXME!
```
Comment (davidtwco) [inline compiler/rustc_borrowck/src/region_infer/opaque_types.rs:644]: What needs fixed? :)

**Review by davidtwco — COMMENTED** [review]
This is a really great start! Thanks for the pull request, if you're intending to keep adding to this then we can mark it as a draft until you've added more, let me know.

**Comment by AndyJado** [comment]
> 

yes, please mark it as a draft.

Sorry for the bot stuff, my intention was to wait in line quietly.

**Comment by compiler-errors** [comment]
^ what the heck happened to this PR? cc @davidtwco

**Comment by AndyJado** [comment]
wow, I'm terribly sorry.. Is there anything I should do? 

I think I was pushing to my own fork

**Comment by compiler-errors** [comment]
lol, you probably just need to open a new PR or something.

**Comment by WaffleLapkin** [comment]
@AndyJado can you (in the future) make separate branches for all things you want to add? This way github won't think that there is no commits, because you'll just use a separate branch and won't need to delete commits.

**Comment by davidtwco** [comment]
> wow, I'm terribly sorry.. Is there anything I should do?
> 
> I think I was pushing to my own fork

That's alright, I think @WaffleLapkin's suggestion is correct, we've got some documentation on how we use Git that you might find helpful:

https://rustc-dev-guide.rust-lang.org/git.html

Just open a new PR with your changes :)

**Pull request merged**